File: ws/js/pilight/submaster/Channel.js
Ext.define('pilight.submaster.Channel', {
    mixins:{
        observable:'Ext.util.Observable'
    },
    constructor:function(config){
        this.mixins.observable.constructor.call(this, config);

        this.addEvents('valueChanged');
    },
    setValue:function(value, callback){
        this.submaster.setValue(this, value, callback);
    },
    getValue:function(value){
        return this.value;
    },
    updateValue:function(value){
        this.value = value;
        this.fireEvent('valueChanged', value);
    }
});

File: android/app/src/main/java/de/justjakob/pilight/control/submaster/Channel.java
package de.justjakob.pilight.control.submaster;

import android.util.Log;

import de.justjakob.pilight.control.SubMaster;

public class Channel {
    private final SubMaster subMaster;

    public String getName() {
        return name;
    }

    public int getValue() {
        return value;
    }

    private String name;
    private int value;

    public Channel(SubMaster subMaster, String name, int value) {
        this.subMaster = subMaster;
        this.name = name;
        this.value = value;
    }

    public void setValue(int value) {
        this.value = value;
        subMaster.setChannelValue(this, value);
    }
}
